protocol/messenger.go

```diff
@@ -739,6 +739,8 @@ func (m *Messenger) ToForeground() {
 	if m.httpServer != nil {
 		m.httpServer.ToForeground()
 	}
+
+	m.asyncRequestAllHistoricMessages()
```

Comment:
Doesn't hurt to confirm, but is this call idempotent? If the messenger goes to the foreground and another request is already inflight, it won't do anything because `m.shouldSync()` would return false?